stdlib/array.ml

```diff
@@ -142,19 +142,22 @@ let of_list = function
         | hd::tl -> unsafe_set a i hd; fill (i+1) tl in
       fill 1 tl
 
-let fold_left f x a =
-  let r = ref x in
-  for i = 0 to length a - 1 do
-    r := f !r (unsafe_get a i)
-  done;
-  !r
+let fold_left f accu a =
```

Comment:
What's the reason for the change?  I believe the new version would always allocate a closure for the inner function, while the old version doesn't allocate anything itself.